contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -3370,25 +3405,52 @@ public String getLTIAccess(String taskId, String contextId){
 		}
 		return ltiUrl;
 	}
+
+	public String getLTIReportAccess(ContentReviewItem item)
+	{
+		String ltiReportsUrl = null;
+		String contentId = item.getContentId();
+		String assignmentId = item.getTaskId();
+		String siteId = item.getSiteId();
+		try
+		{
+			String ltiReportsId = siteService.getSite(siteId).getProperties().getProperty("turnitin_reports_lti_id");
+			ContentResource resource = null;
+			resource = contentHostingService.getResource(contentId);
+			ResourceProperties resourceProperties = resource.getProperties();
+			String ltiResourceId = resourceProperties.getProperty("turnitin_id");
+			if (ltiResourceId == null)
+			{
+				// Fallback: link to assignment
+				return getLTIAccess(assignmentId, siteId);
+			}
+			ltiReportsUrl = "/access/basiclti/site/" + siteId + "/content:" + ltiReportsId + ",resource:" + ltiResourceId;
```

Comment:
I think this requires more changes to the BasicLTI code to support this new URL and wonder if we should actually be having our own launch URLs handled by TII under `/access/turnitin/....` and then we can create custom launches (still using the data from the central LTI registry) but don't have to add special case handling specifically for Turnitin. What do you think?